test/cmp_msg_test.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * Copyright 2007-2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright Nokia 2007-2019
+ * Copyright Siemens AG 2015-2019
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "cmp_testlib.h"
+
+static const char *server_cert_f;
+static const char *pkcs10_f;
+
+typedef struct test_fixture {
+    const char *test_case_name;
+    OSSL_CMP_CTX *cmp_ctx;
+    /* for msg create tests */
+    int bodytype;
+    int err_code;
+    /* for certConf */
+    int fail_info;
+    /* for protection tests */
+    OSSL_CMP_MSG *msg;
+    int expected;
+    /* for error and response messages */
+    OSSL_CMP_PKISI *si;
+} CMP_MSG_TEST_FIXTURE;
+
+static unsigned char ref[CMP_TEST_REFVALUE_LENGTH];
+
+static void tear_down(CMP_MSG_TEST_FIXTURE *fixture)
+{
+    OSSL_CMP_CTX_free(fixture->cmp_ctx);
+    OSSL_CMP_MSG_free(fixture->msg);
+    OSSL_CMP_PKISI_free(fixture->si);
+    OPENSSL_free(fixture);
+}
+
+#define SET_OPT_UNPROTECTED_SEND(ctx, val) \
+    OSSL_CMP_CTX_set_option((ctx), OSSL_CMP_OPT_UNPROTECTED_SEND, (val))
+static CMP_MSG_TEST_FIXTURE *set_up(const char *const test_case_name)
+{
+    CMP_MSG_TEST_FIXTURE *fixture;
+
+    if (!TEST_ptr(fixture = OPENSSL_zalloc(sizeof(*fixture))))
+        return NULL;
+    fixture->test_case_name = test_case_name;
+
+    if (!TEST_ptr(fixture->cmp_ctx = OSSL_CMP_CTX_new())
+            || !TEST_true(SET_OPT_UNPROTECTED_SEND(fixture->cmp_ctx, 1))
+            || !TEST_true(OSSL_CMP_CTX_set1_referenceValue(fixture->cmp_ctx,
+                                                           ref, sizeof(ref)))) {
+        tear_down(fixture);
+        return NULL;
+    }
+    return fixture;
+}
+
+static EVP_PKEY *newkey = NULL;
+static X509 *cert = NULL;
+
+#define EXECUTE_MSG_CREATION_TEST(expr) \
+do { \
```

Comment:
should this be indented?

Reply:
Yes indeed.
I actually had added a workaround in the new format checking tool: https://github.com/openssl/openssl/pull/10363 for situations like this because there quite many lines in our CMP tests where macro bodies should be indented by +4.
Due to your comment I'll make the tool by default complain about such issues.

Reply:
Yes indeed.
I actually had added a workaround in the new format checking tool: #10363 for situations like this because there are >200 lines in our CMP tests where macro bodies should be indented by +4.
Due to your comment I'll make the tool by default complain about such issues.

Reply:
I fixed this indentation and also some other issues found by the new format checking tool: #10363